src/history.rs

```diff
@@ -219,17 +225,35 @@ impl<'a> DynamicHistory<'a> {
         }
     }
 
-    fn retreive_next_persistent_epoch(&mut self) {
-        if self.restart_counter.is_tracking_restarts() {
-            match self.restart_counter.next_sequence_id(self.probe_id) {
-                Some(next_persistent_epoch) => {
-                    self.self_clock.epoch = next_persistent_epoch.get().into()
-                }
-                None => {
-                    self.self_clock.epoch = 0.into();
-                    self.record_event(EventId::EVENT_INVALID_NEXT_EPOCH_SEQ_ID);
-                }
+    /// Isolated function for figuring out what the next epoch should be for the probe.
+    fn calculate_next_epoch(
+        restart_counter: &mut RestartSequenceCounter,
+        probe_id: ProbeId,
+        prior_epoch: Option<ProbeEpoch>,
+    ) -> (ProbeEpoch, RestartCounterProvidedInvalidEpochSeqId) {
+        if restart_counter.is_tracking_restarts() {
+            if let Some(next_persistent_sequence_epoch) = restart_counter.next_sequence_id(probe_id)
+            {
+                (
+                    ProbeEpoch(next_persistent_sequence_epoch.get()),
+                    RestartCounterProvidedInvalidEpochSeqId(false),
+                )
+            } else {
+                (
+                    ProbeEpoch::MIN,
+                    RestartCounterProvidedInvalidEpochSeqId(true),
+                )
             }
+        } else if let Some(prior_epoch) = prior_epoch {
+            (
+                ProbeEpoch(core::cmp::max(1, prior_epoch.0.wrapping_add(1))),
```

Comment:
Shouldn't this one go away too?

Reply:
that is, I think this should just be regular wrapping add, now that the `1` epoch is no longer special.

Reply:
Addressed in the latest commit